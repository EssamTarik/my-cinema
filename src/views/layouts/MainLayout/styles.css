.main-layout {
  display: flex;
  flex-direction: column;

  overflow: hidden;

  height: 100%;
  width: 100%;

  background-color: #0A1016;
}
.main-layout__content {
  overflow: auto;
  flex: 1;
}
