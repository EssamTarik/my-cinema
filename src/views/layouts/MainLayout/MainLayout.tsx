import React from 'react';
import { NavigationBar } from '../../../components';
import './styles.css';

const MainLayout = ({ children }: React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>) => (
  <div className='main-layout'>
    <NavigationBar />
    {children}
  </div>
)

export default MainLayout;