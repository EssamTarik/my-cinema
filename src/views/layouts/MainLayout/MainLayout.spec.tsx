import React from 'react';
import MainLayout from './MainLayout';
import { render, cleanup } from '@testing-library/react';
jest.mock('../../../components', () => ({ NavigationBar: () => <div /> }))

describe('MainLayout', () => {
  afterEach(cleanup);

  it('renders', () => {
    const { asFragment } = render(<MainLayout />);
    expect(asFragment()).toMatchSnapshot()
  })
})