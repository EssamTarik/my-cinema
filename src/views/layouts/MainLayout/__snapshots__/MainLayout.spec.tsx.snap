// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders 1`] = `
<DocumentFragment>
  <div
    class="main-layout"
  >
    <div />
  </div>
</DocumentFragment>
`;
